src/databricks/labs/remorph/reconcile/query_builder.py

```diff
@@ -163,10 +151,127 @@ def _construct_hash_query(table_name: str, query_filter: str, hash_expr: str, ke
+                transformation = self._get_layer_transform(transformation_dict, column, self.layer)
+            else:
+                transformation = None
+
+            column_origin, column_alias = self._get_column_alias(self.layer, column, column_mapping)
+
+            transformation_rule_mapping.append(TransformRuleMapping(column_origin, transformation, column_alias))
+
+        return transformation_rule_mapping
+
+    def _get_default_transformation(self, columns, column_mapping, schema):
+        transformation_rule_mapping = []
+        for column in columns:
+            column_origin = column if self.layer == "source" else self._get_column_map(column, column_mapping)
+            column_data_type = schema.get(column_origin).data_type
+            transformation = self._get_default_transformation_expr(self.source, column_data_type).format(column_origin)
+
+            column_origin, column_alias = self._get_column_alias(self.layer, column, column_mapping)
+
+            transformation_rule_mapping.append(TransformRuleMapping(column_origin, transformation, column_alias))
+
+        return transformation_rule_mapping
+
+    @staticmethod
+    def _get_default_transformation_expr(data_source: str, data_type: str) -> str:
+        if data_source == SourceType.ORACLE.value:
+            return OracleDataSource.oracle_datatype_mapper.get(data_type, ColumnTransformationType.ORACLE_DEFAULT.value)
+        if data_source == SourceType.SNOWFLAKE.value:
+            return SnowflakeDataSource.snowflake_datatype_mapper.get(
+                data_type, ColumnTransformationType.SNOWFLAKE_DEFAULT.value
+            )
+        if data_source == SourceType.DATABRICKS.value:
+            return DatabricksDataSource.databricks_datatype_mapper.get(
+                data_type, ColumnTransformationType.DATABRICKS_DEFAULT.value
+            )
+        msg = f"Unsupported source type --> {data_source}"
+        raise ValueError(msg)
+
+    @staticmethod
+    def _get_column_alias(layer, column, column_mapping):
+        if column_mapping and column in column_mapping.keys() and layer == "target":
+            column_alias = column_mapping.get(column).source_name
+            column_origin = column_mapping.get(column).target_name
+        else:
+            column_alias = column
+            column_origin = column
+
+        return column_origin, column_alias
+
+    def build_threshold_query(self) -> str:
+        column_mapping = self.table_conf.list_to_dict(ColumnMapping, "source_name")
+        transformations_dict = self.table_conf.list_to_dict(Transformation, "column_name")
+
+        threshold_columns = set(threshold.column_name for threshold in self.table_conf.thresholds)
+        join_columns = set(self.table_conf.join_columns)
+
+        if self.table_conf.jdbc_reader_options and self.layer == "source":
+            partition_column = {self.table_conf.jdbc_reader_options.partition_column}
+        else:
+            partition_column = set()
```

Comment:
These column's identification should be solely done in one function which we already have _get_column_list

Reply:
fixed.